ensime_shared/ensime.py

```diff
@@ -356,15 +356,31 @@ def handle_symbol_info(self, call_id, payload):
                 self.set_position(decl_pos)
                 del self.call_options[call_id]
 
+    def handle_new_scala_notes_event_with_syntastic(self, call_id, payload):
+        """Syntastic specific handler for response `NewScalaNotesEvent`."""
+
+        current_file = os.path.abspath(self.path())
+        loclist = list({
+                'bufnr': self.vim.current.buffer.number,
+                'lnum': note['line'],
+                'col': note['col'],
+                'text': note['msg'],
+                'len': note['end'] - note['beg'] + 1,
+                'type': note['severity']['typehint'][4:5],
+                'valid': 1
+            } for note in payload["notes"] if current_file == os.path.abspath(note['file']))
+        self.vim.command('let b:ensime_scala_notes += {}'.format(json.dumps(loclist)))
```

Comment:
Same here. Please do the same for any 'vim' command, be it eval or command itself.